src/Authenticator/LoginHandler.php

```diff
@@ -99,18 +100,45 @@ public function getSchema()
         }
 
         // Get a list of authentication for the user and the find default
-        $authMethods = $member->RegisteredMFAMethods();
+        $registeredMethods = $member->RegisteredMFAMethods();
 
         // Pool a list of "lead in" labels. We skip the default here assuming it's not required.
         $alternateLeadInLabels = [];
-        foreach ($authMethods as $method) {
-            $alternateLeadInLabels[str_replace('\\', '-', get_class($method))] =
+        foreach ($registeredMethods as $method) {
+            $alternateLeadInLabels[$method->getMethod()->getURLSegment()] =
                 $method->getLoginHandler()->getLeadInLabel();
         }
 
-        return [
-            'registered_methods' => $alternateLeadInLabels,
-        ];
+        // Prepare an array to hold details for available methods to register
+        $registrationDetails = [];
+        $registeredMethodNames = array_keys($alternateLeadInLabels);
+
+        // Get all methods that may be registered
+        $allMethods = MethodRegistry::singleton()->getAllMethods();
+
+        // Resolve details for methods that aren't setup
+        foreach ($allMethods as $method) {
+            // Skip registration details if the user has already registered this method
+            if (in_array($method->getURLSegment(), $registeredMethodNames)) {
+                continue;
+            }
+
+            $registerHandler = $method->getRegisterHandler();
+
+            $registrationDetails[$method->getURLSegment()] = [
+                'name' => $registerHandler->getName(),
+                'description' => $registerHandler->getDescription(),
+                'supportLink' => $registerHandler->getSupportLink(),
+            ];
+        }
+
+        $defaultMethod = $member->DefaultRegisteredMethod;
```

Comment:
You mentioned you wanted to make this a has one. If you wrap it in a model method e.g. `getDefaultRegisteredMethod()` now then you won't have to change this later

Reply:
Wouldn't `DefaultRegisteredMethod` still be suitable for the `has_one`? In the end I opted not to do `has_one` because there's no good ORM way (that I know of) to ensure it's the same instance whether you get it from the `has_one` or the `has_many`.

Reply:
Is this field called DefaultRegisteredMethodID now?

Reply:
`$member->getDefaultRegisteredMethod()`?

Reply:
Yeah the field is called `DefaultRegisteredMethodID` and the `getDefaultRegisteredMethod` is an accessor for `DefaultRegisteredMethod` - which `DataObject` magically scaffolds.

Do you think I should use the accessor directly? I can but I feel like for `DataObject` specifically that's not a very common flow.

Reply:
If you think it's OK and it's not a mistake, then all good - it just reads like one